libpysal/weights/raster.py

```diff
@@ -128,6 +128,60 @@ def da2W(
     w.index = wsp.index
     return w
 
+def get_nodata(da):
+    """
+    Identify nodata value in a `DataArray`
+
+    NOTE: follows guidance from https://corteva.github.io/rioxarray/stable/getting_started/nodata_management.html
+    ...
+
+    Arguments
+    ---------
+    da : xarray.DataArray
+        Input 2D or 3D DataArray with shape=(z, y, x)
+
+    Returns
+    -------
+    nodata : int/float
+            Value used for nodata pixels. If no value is available, `None` is
+            returned
+    """
+    try:
+        return da.rio.nodata
+    except:
+        nodata = nodata_from_attrs(da.attrs)
+        if nodata is not None:
+            return nodata
+        return None
```

Comment:
This is redundant, isn't? 
```suggestion
        return nodata_from_attrs(da.attrs)
```

If it is `None` it returns `None`, if not it returns a value.